psycop/common/model_training_v2/problem_type/eval_dataset_base.py

```diff
@@ -4,27 +4,41 @@
 
 import polars as pl
 
-from psycop.common.model_training_v2.metrics.binary_metrics.base import BinaryMetric
-
-from ..presplit_preprocessing.polars_frame import PolarsFrame
+from psycop.common.global_utils.pickle import write_to_pickle
+from psycop.common.model_training_v2.metrics.base_metric import CalculatedMetric
+from psycop.common.model_training_v2.metrics.binary_metrics.base_binary_metric import (
+    BinaryMetric,
+)
 
 
 @dataclass
 class BaseEvalDataset:
-    pred_time_uuids: pl.Expr
-    y_hat_probs: pl.Expr
-    y: pl.Expr
-    df: PolarsFrame
+    pred_time_uuids: str
+    y_hat_probs: str
+    y: str
+    df: pl.DataFrame
 
     def to_disk(self, path: Path) -> None:
         ...
 
 
-class BinaryEvalDataset(pl.DataFrame):
-    pred_time_uuids: pl.Expr
-    y_hat_probs: pl.Expr
-    y: pl.Expr
-    df: PolarsFrame
+class BinaryEvalDataset(BaseEvalDataset):
+    pred_time_uuids: str
+    y_hat_probs: str
+    y: str
+    df: pl.DataFrame
+
+    def calculate_metrics(
+        self,
+        metrics: Iterable[BinaryMetric],
+    ) -> list[CalculatedMetric]:
+        return [
+            metric.calculate(
+                y_true=self.df.get_column(self.y),
+                y_pred=self.df.get_column(self.y_hat_probs),
+            )
+            for metric in metrics
+        ]
 
-    def calculate_metrics(self, metrics: Iterable[BinaryMetric]) -> dict[str, float]:
-        ...
+    def to_disk(self, path: Path) -> None:
+        write_to_pickle(self, filepath=path)
```

Comment:
We definitely do not want to pickle this. Will break all the time. Rather, the disk-interface should just be parquet or whatever. Probably want users to decide here.